numpy/lib/npyio.py

```diff
@@ -178,6 +178,9 @@ class NpzFile(Mapping):
     array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
 
     """
+    # Make __exit__ safe if zipfile_factory raises an exception
+    zip = None
+    fid = None
```

Comment:
You're right that this makes them static, but it only makes the initial value static - which does exactly what we need, guaranteeing they are always set.